Client/Telemetry/TelemetryWrapper.swift

```diff
@@ -135,8 +137,137 @@ class UnifiedTelemetry {
             return outputDict
         }
 
-       Telemetry.default.add(pingBuilderType: CorePingBuilder.self)
-       Telemetry.default.add(pingBuilderType: MobileEventPingBuilder.self)
+        legacyTelemetry.add(pingBuilderType: CorePingBuilder.self)
+        legacyTelemetry.add(pingBuilderType: MobileEventPingBuilder.self)
+
+        // Initialize Glean
+        initGlean(profile, sendUsageData: sendUsageData)
+    }
+
+    func initGlean(_ profile: Profile, sendUsageData: Bool) {
+        // Get the legacy telemetry ID and record it in Glean for the deletion-request ping
+        if let uuidString = UserDefaults.standard.string(forKey: "telemetry-key-prefix-clientId"), let uuid = UUID(uuidString: uuidString) {
+            GleanMetrics.LegacyIds.clientId.set(uuid)
+        }
+
+        // Initialize Glean telemetry
+        glean.initialize(uploadEnabled: sendUsageData, configuration: Configuration(channel: AppConstants.BuildChannel.rawValue))
+
+        // Save the profile so we can record settings from it when the notification below fires.
+        self.profile = profile
+
+        // Register an observer to record settings and other metrics that are more appropriate to
+        // record on closing rather than during initialization.
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(appDidEnterBackground(notification:)),
+            name: UIApplication.didEnterBackgroundNotification,
+            object: nil
+        )
+    }
+
+    @objc func appDidEnterBackground(notification: NSNotification) {
+        // We're going to background, so store record pertinent metrics
+        if let profile = self.profile {
+            // Record default search engine setting
+            let searchEngines = SearchEngines(prefs: profile.prefs, files: profile.files)
+            GleanMetrics.Search.defaultEngine.set(searchEngines.defaultEngine.engineID ?? "custom")
+
+            let prefs = profile.prefs
+            // Record New Tab setting
+            if let newTabChoice = prefs.stringForKey(NewTabAccessors.HomePrefKey) {
+                GleanMetrics.Preferences.newTabExperience.set(newTabChoice)
+            }
+            // Record chosen email client setting
+            if let chosenEmailClient = prefs.stringForKey(PrefsKeys.KeyMailToOption) {
+                GleanMetrics.Preferences.mailClient.set(chosenEmailClient)
+            }
+        }
+
+        // Record the open tab count
+        let delegate = UIApplication.shared.delegate as? AppDelegate
+        if let count = delegate?.tabManager.count {
+            GleanMetrics.Tabs.openCount.add(Int32(count))
+        }
+
+        // Record other preferences stored in UserDefaults
+        if let userDefaults = UserDefaults(suiteName: Telemetry.default.configuration.userDefaultsSuiteName) {
```

Comment:
Aside: I need to understand why `Telemetry.default.configuration.userDefaultsSuiteName` is in this class, as profile.prefs should be all that is needed.

Prefer: 
```
            let keys = profile.prefs.keys() // not sure how heavy this is, best to get it once
            func getPref(_ name: String, default: Bool) {  /*handy local func to get pref or default val*/ }
            GleanMetrics.Preferences.blockPopups.set(getPref("blah", default: true))
            GleanMetrics.Preferences.anotherOne.set(getPref(....))
```

NSUserDefaultPrefs would need a keys:
```
 open func keys() -> [String] { return Array(userDefaults.dictionaryRepresentation().keys) }
```

EDIT: argh, Prefs is a protocol, also need MockProfilePrefs to have `open func keys() -> [String] { return things.keyEnumerator().allObjects }`

Reply:
Thanks, I didn't dig into the `Profile` to know that everything was in there.  Instead, I was following how legacy telemetry was capturing the values.  Let me refactor this and see if I can get everything out of the `Profile` instead.

Reply:
Hopefully I am not sending you down a garden path, even if you want to land as-is I could follow up with some of this nitpicky stuff in a follow-up PR and have you review it. My assumptions about 'this change looks easy' is often wrong in our codebase ;).

Reply:
Let me give it a shot and see, I would prefer a cleaner approach if possible and it sounds like this may help.

Reply:
Okay, I think I managed to find all of the prefs that were being recorded in the `Profile.Prefs` rather than scraping them from the telemetry store.  The theme prefs I was able to get from the `ThemeManager`.